centreon-certified/signl4/signl4-events-apiv2.lua

```diff
@@ -191,85 +189,77 @@ function EventQueue:add()
-      }
-    )
-    -- set proxy address configuration
-    if (self.sc_params.params.proxy_address ~= '') then
-      if (self.sc_params.params.proxy_port ~= '') then
-        http_request:setopt(curl.OPT_PROXY, self.sc_params.params.proxy_address .. ':' .. self.sc_params.params.proxy_port)
-      else
-        self.sc_logger:error("[EventQueue:send_data]: proxy_port parameter is not set but proxy_address is used")
-      end
+  :setopt_url(self.sc_params.params.server_address .. "/webhook/" .. self.sc_params.params.team_secret)
+  :setopt_writefunction(
+    function (response)
+      http_response_body = http_response_body .. tostring(response)
     end
-
-    -- set proxy user configuration
-    if (self.sc_params.params.proxy_username ~= '') then
-      if (self.sc_params.params.proxy_password ~= '') then
-        http_request:setopt(curl.OPT_PROXYUSERPWD, self.sc_params.params.proxy_username .. ':' .. self.sc_params.params.proxy_password)
-      else
-        broker_log:error("[EventQueue:send_data]: proxy_password parameter is not set but proxy_username is used")
-      end
+  )
+  :setopt(curl.OPT_TIMEOUT, self.sc_params.params.connection_timeout)
+  :setopt(curl.OPT_SSL_VERIFYPEER, self.sc_params.params.allow_insecure_connection)
+  :setopt(
+    curl.OPT_HTTPHEADER,
+    {
+      "content-type: application/json;charset=UTF-8",
+      "content-length: " .. string.len(http_post_data)
+    }
+  )
+  -- set proxy address configuration
+  if (self.sc_params.params.proxy_address ~= '') then
+    if (self.sc_params.params.proxy_port ~= '') then
+      http_request:setopt(curl.OPT_PROXY, self.sc_params.params.proxy_address .. ':' .. self.sc_params.params.proxy_port)
+    else
+      self.sc_logger:error("[EventQueue:send_data]: proxy_port parameter is not set but proxy_address is used")
     end
-
-    -- adding the HTTP POST data
-    http_request:setopt_postfields(http_post_data)
-
-    -- performing the HTTP request
-    http_request:perform()
-
-    -- collecting results
-    http_response_code = http_request:getinfo(curl.INFO_RESPONSE_CODE)
-
-    http_request:close()
-
-    -- Handling the return code
-    local retval = false
-    if http_response_code == 200 then
-      self.sc_logger:info("[EventQueue:send_data]: HTTP POST request successful: return code is " .. tostring(http_response_code))
-      retval = true
+  end
+  -- set proxy user configuration
+  if (self.sc_params.params.proxy_username ~= '') then
+    if (self.sc_params.params.proxy_password ~= '') then
+      http_request:setopt(curl.OPT_PROXYUSERPWD, self.sc_params.params.proxy_username .. ':' .. self.sc_params.params.proxy_password)
     else
-      self.sc_logger:error("[EventQueue:send_data]: HTTP POST request FAILED, return code is " .. tostring(http_response_code) .. ". Message is: " .. tostring(http_response_body))
+      broker_log:error("[EventQueue:send_data]: proxy_password parameter is not set but proxy_username is used")
```

Comment:
why broker_log ?

Reply:
![parce-que-orangina](https://user-images.githubusercontent.com/12429463/143553366-ab2d3682-0504-4bfa-a673-c58a1ae4090c.gif)

Should be ok now ;)